### ISSUE TITLE
Rename to 'WindowActivityTracker'

### ISSUE THREAD

**Issue by AlexanderLill** [HASEL-UZH/PA.WindowsActivityTracker#4]
This tracker tracks window activity, and to avoid misunderstandings with the operating system "Windows" it might be worth renaming it. This clarifies that the tracker also works on non-windows systems.

**Comment by royru** [comment]
Good point, we should probably do that (@casaout thoughts?). We need to be a bit careful since lots of repos use this as a git submodule. Would probably make sense to work out a "migration guide" for these affected repos.

**Comment by casaout** [comment]
I agree that the part `Windows` in `WindowsActivityTracker` might be confusing and is suboptimal. It started with the idea that several windows' activities result in one higher-level activity (e.g. you're researching something on different websites/tabs that can be grouped into `work-related web browsing`).

The tracker is actually doing several things:
- tracks current app usage (and window title)
- maps the app names and window titles to an activity (based on the TSE'17) paper

So maybe, we can think of an entirely better name? (e.g. AppActivityTracker)

Also, changing the name might require a couple of refactorings, if we're doing this properly - it's used in several student projects, some of ours, and (the C# and Swift versions) in the public [PersonalAnalytics](https://github.com/HASEL-UZH/PersonalAnalytics) repo.

**Comment by kevin-chow** [comment]
Would just `ActivityTracker` work? It seems like the high level goal is categorizing window titles/app usage/browser URLs into a higher-level activity labels (including being idle). But the downside is "Activity" by itself might not be descriptive enough. 

Another option is should we even keep the suffix "Tracker?" Would `ActivityClassifier` work, since that kind of is what it is doing (albeit using a very simple algorithm)? But the downside of that too is that it could mislead people into thinking there is some ML component going on.

**Comment by casaout** [comment]
It depends quite a bit on how it's actually used in the end. Many projects never use the activity categories, but just the "raw" data which is the start + end time and window title of a process that was in the foreground.

I agree, the `tracker` suffix is not ideal; it might have a bit of a creepy connotation to it. Today, I'd prefer `monitor`. The reason is that it helps to uniformly identify all the different trackers/monitors easily in the code. 

Note that the WindowsActivityTracker does not just perform the mapping, but also collect the actual raw data through OS hooks.
